### PR TITLE
feat(components): show unknown date only in bar chart and table in pevalence-over-time

### LINKED ISSUE #272: Remove unknown from prevalence over time line chart
In the line chart, it doesn't make sense to have "Unknown":

image

For the other plots and for the table, it should stay.

### PR DESCRIPTION
**Pull request by JonasKellerer**
resolves #272

### Summary

When the dateRange is null, then unknown now shown on the table and on the bar chart. On the line and bubble chart these entries are removed.

### Screenshot

grafik

grafik

### PR Checklist

~~- [ ] All necessary documentation has been adapted.~~
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

File: components/src/preact/prevalenceOverTime/prevalence-over-time-bar-chart.tsx [hunk 1]
```diff
@@ -25,14 +26,24 @@ const PrevalenceOverTimeBarChart = ({
...
     const maxY =
-        yAxisScaleType !== 'logit' ? getYAxisMax(maxInData(data), yAxisMaxConfig?.[yAxisScaleType]) : undefined;
+        yAxisScaleType !== 'logit'
+            ? getYAxisMax(maxInData(nullFirstData), yAxisMaxConfig?.[yAxisScaleType])
+            : undefined;
 
     const config: ChartConfiguration = {
         type: BarWithErrorBarsController.id,
         data: {
-            labels: data[0]?.content.map((dateRange) => dateRange.dateRange?.toString() ?? 'Unknown') || [],
```
Comment (fengelniederhammer) [inline components/src/preact/prevalenceOverTime/prevalence-over-time-bar-chart.tsx:34]: What did those `labels` do? Apparently nothing, if we can simply delete them?

File: components/src/preact/prevalenceOverTime/prevalence-over-time-line-chart.tsx [hunk 2]
```diff
@@ -100,13 +115,29 @@ const getDatasetCILower = (prevalenceOverTimeVariant: PrevalenceOverTimeVariantD
...
+    data: prevalenceOverTimeVariant.content.map((dataPoint): Datapoint => {
+        const ciLimits = wilson95PercentConfidenceInterval(dataPoint.count, dataPoint.total);
+
+        return {
+            y: dataPoint.prevalence,
+            x: dataPoint.dateRange?.toString(),
+            yCiUpper: ciLimits.upperLimit,
+            yCiLower: ciLimits.lowerLimit,
+        };
+    }),
```
Comment (fengelniederhammer) [inline components/src/preact/prevalenceOverTime/prevalence-over-time-line-chart.tsx:127]: Why did this change?

**Review by fengelniederhammer — APPROVED** [review]

File: components/src/preact/prevalenceOverTime/prevalence-over-time-bar-chart.tsx (hunk 1 above)
Comment (JonasKellerer) [reply components/src/preact/prevalenceOverTime/prevalence-over-time-bar-chart.tsx:34]: They are now put into the dataset. They are used as x labels. They must have the same length as the dataset But now the dataset contains the full datapoint. I thought this is more expressive and easier to sort together the x and y value.

File: components/src/preact/prevalenceOverTime/prevalence-over-time-line-chart.tsx (hunk 2 above)
Comment (JonasKellerer) [reply components/src/preact/prevalenceOverTime/prevalence-over-time-line-chart.tsx:127]: So we have access to the value in the tooltip, without the need to get the other dataset and find the value with the corresponding index.

**Pull request merged**